python/packages/ai/teams/ai/planner/openai_planner.py

```diff
@@ -3,32 +3,39 @@
 Licensed under the MIT License.
 """
 from datetime import datetime
-from typing import Any, Callable, Optional, Union
+from typing import Any, Callable, List, Tuple, Union
 
 import semantic_kernel as sk
 from botbuilder.core import TurnContext
-from semantic_kernel.connectors.ai import CompleteRequestSettings
+from semantic_kernel.connectors.ai import (
+    ChatCompletionClientBase,
+    ChatRequestSettings,
+    CompleteRequestSettings,
+    TextCompletionClientBase,
+)
 from semantic_kernel.connectors.ai.open_ai import (
     OpenAIChatCompletion,
     OpenAITextCompletion,
 )
-from semantic_kernel.connectors.ai.text_completion_client_base import (
-    TextCompletionClientBase,
-)
-from semantic_kernel.utils.null_logger import NullLogger
 
 from teams.ai import AIHistoryOptions
 from teams.ai.prompts import PromptManager
 from teams.ai.prompts.prompt_template import PromptTemplate
 from teams.ai.prompts.utils import generate_sk_prompt_template_config
-from teams.ai.state import TurnState
+from teams.ai.state import ConversationT, TempT, TurnState, UserT
 
 from .command_type import CommandType
+from .conversation_history import state_to_history_array
 from .openai_planner_options import OpenAIPlannerOptions
 from .plan import Plan
 from .planner import Planner
 from .response_parser import parse_response
 
+# Define role names used by OpenAI API
+SYSTEM_ROLE_NAME: str = "system"
+USER_ROLE_NAME: str = "user"
+ASSISTANT_ROLE_NAME: str = "assistant"
+
 
 class OpenAIPlanner(Planner):
```

Comment:
Also I noticed you logging requests using `print`, the `AI` class has a logger that can be passed in here for use. This is preferred because we want the user to be able to pass in their own logger instance if they want.

Reply:
Created task https://github.com/microsoft/teams-ai/issues/445